conda_smithy/configure_feedstock.py

```diff
@@ -1842,11 +1843,22 @@ def _load_forge_config(forge_dir, exclusive_config_file, forge_yml=None):
             )
         )
 
-    for platform_arch in config["build_platform"].keys():
+    build_platforms = sorted(config["build_platform"].keys())
+
+    for platform_arch in build_platforms:
         config[platform_arch] = {"enabled": "True"}
         if platform_arch not in config["provider"]:
             config["provider"][platform_arch] = None
 
+    config["noarch_platform"] = conda_build.utils.ensure_list(config["noarch_platform"])
+
+    for noarch_platform in sorted(config["noarch_platform"]):
+        if noarch_platform not in build_platforms:
```

Comment:
We don't need the `noarch_platform` to be one of `build_platforms`. We need it to be one of the platforms we build for.
For example, `noarch_platform` can be `linux_aarch64`, but we cross compile `linux_aarch64` in `linux_64` and therefore `linux_aarch64` is not in `build_platforms`.

Reply:
never mind. I was confused with the naming here

Reply:
so, to be clear: should i be looking in the keys, or the values for valid  noarch_platforms?